src/components/role-review-card.tsx

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable @next/next/no-img-element */
+import { cn } from "~/lib/utils";
+import {
+  Card,
+  CardContent,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "~/components/ui/card";
+import { api } from "~/trpc/server";
+import { ReviewCardStars } from "./review-card-stars";
+
+// TODO: move this somewehre else?
+function formatDate(date: Date) {
+  const months = [
+    "Jan",
+    "Feb",
+    "Mar",
+    "Apr",
+    "May",
+    "Jun",
+    "Jul",
+    "Aug",
+    "Sep",
+    "Oct",
+    "Nov",
+    "Dec",
+  ];
+
+  // Extract the year, month, and day components
+  const year = date.getFullYear();
+  const monthIndex = date.getMonth();
+  const month = months[monthIndex];
+  const day = String(date.getDate()).padStart(2, "0"); // Ensure two digits for the day
+
+  // Construct the formatted date string
+  const formattedDate = `${day} ${month} ${year}`;
+
+  return formattedDate;
+}
+
+// todo: add this attribution in a footer somewhere
+//  <a href="https://clearbit.com">Logos provided by Clearbit</a>
+
+type RoleReviewCardProps = {
+  className?: string;
+  roleObj: {
+    id: string;
+    title: string;
+    description: string | null;
+    companyId: string;
+  };
+};
+
+export async function RoleReviewCard({
+  className,
+  roleObj,
+  ...props
+}: RoleReviewCardProps) {
+  // ===== COMPANY DATA ===== //
+  const company = await api.company.getById.query({ id: roleObj.companyId });
+  const roleDescription =
+    roleObj.description && roleObj.description.length >= 210
+      ? cn(roleObj.description.slice(0, 210), "...")
+      : roleObj.description;
+  const positionDate: string = formatDate(company.createdAt);
+
+  // ===== REVIEW DATA ===== //
+  const reviews = await api.review.getByRole.query({ roleId: roleObj.id });
+  const reviewCount = reviews.length;
+  let workEnvironment = "In Person";
+  // TODO: we are setting the work enviornment to the first role since its not stored per company. Change?
```

Comment:
Yikes, right ... Should we update the role schema to include the work environment? We can maybe talk this through during standup. However, I think it should be safe to assign a work environment to the role rather than individual reviews.